core/domain/stats_domain_test.py

```diff
@@ -599,29 +872,78 @@ def test_update_exp_issue_from_model(self):
+
+        with convert_issue_dict_swap, current_issue_schema_version_swap:
+            exp_issue_from_model = stats_services.get_exp_issues_from_model(
+                exp_issues_model)
+
+        self.assertEqual(
+            exp_issue_from_model.unresolved_issues[0].issue_type, 'EarlyQuit1')
         self.assertEqual(
-            exp_issue_dict['issue_customization_args']['new_key'], 5)
+            exp_issue_from_model.unresolved_issues[0].issue_customization_args[
+                'new_key'], 5)
 
         # For other issue types, no changes happen during migration.
         exp_issue1 = stats_domain.ExplorationIssue(
             'MultipleIncorrectSubmissions', {}, [], 1, True)
         exp_issue_dict1 = exp_issue1.to_dict()
-        with self.swap(
+
+        stats_models.ExplorationIssuesModel.create(
+            'exp_id_1', 1, [exp_issue_dict1])
+
+        exp_issues_model1 = stats_models.ExplorationIssuesModel.get_model(
+            'exp_id_1', 1)
+
+        current_issue_schema_version_swap = self.swap(
+            stats_models, 'CURRENT_ISSUE_SCHEMA_VERSION', 2)
+        convert_issue_dict_swap = self.swap(
             stats_domain.ExplorationIssue,
             '_convert_issue_v1_dict_to_v2_dict',
-            self._dummy_convert_issue_v1_dict_to_v2_dict):
-            stats_domain.ExplorationIssue.update_exp_issue_from_model(
-                exp_issue_dict1)
+            self._dummy_convert_issue_v1_dict_to_v2_dict)
+
+        with convert_issue_dict_swap, current_issue_schema_version_swap:
+            exp_issue_from_model1 = stats_services.get_exp_issues_from_model(
+                exp_issues_model1)
+
         self.assertEqual(
-            exp_issue_dict1['issue_type'], 'MultipleIncorrectSubmissions')
+            exp_issue_from_model1.unresolved_issues[0].issue_type,
+            'MultipleIncorrectSubmissions')
+
+    def test_cannot_update_exp_issue_from_invalid_schema_version_model(self):
+        exp_issue = stats_domain.ExplorationIssue('EarlyQuit', {}, [], 4, True)
+        exp_issue_dict = exp_issue.to_dict()
+        stats_models.ExplorationIssuesModel.create(
+            'exp_id', 1, [exp_issue_dict])
+
+        exp_issues_model = stats_models.ExplorationIssuesModel.get_model(
+            'exp_id', 1)
+
+        with self.assertRaisesRegexp(
+            Exception,
+            'Sorry, we can only process v1-v%d and unversioned issue schemas at'
+            ' present.' %
+            stats_models.CURRENT_ISSUE_SCHEMA_VERSION):
+            stats_services.get_exp_issues_from_model(exp_issues_model)
+
+    def test_actual_update_exp_issue_from_model_raises_error(self):
+        exp_issue = stats_domain.ExplorationIssue('EarlyQuit', {}, [], 1, True)
+        exp_issue_dict = exp_issue.to_dict()
+
+        with self.assertRaises(NotImplementedError):
+            stats_domain.ExplorationIssue.update_exp_issue_from_model(
+                exp_issue_dict)
 
     def test_validate(self):
```

Comment:
Likewise, remove old code.

Reply:
This is not old code. There are some validations here too

Reply:
Ah I see. In that case, split these up so they have similar structure as `test_validate_issue_type`, `test_validate_playthrough_ids`, etc.

Reply:
I think they can be a part of https://github.com/oppia/oppia/issues/6853?

Reply:
Yup that SGTM! Please don't forget, then, to reference these lines in that issue.

This is a good opportunity to resolve Sandeep's [comment](https://github.com/oppia/oppia/issues/6853#issuecomment-500533134) now so adding new validate methods to split up is easier later.